src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -214,17 +213,43 @@ var ReactDOMInput = {
       );
     }
 
+    var node = ReactDOMComponentTree.getNodeFromInstance(inst);
     var value = LinkedValueUtils.getValue(props);
     if (value != null) {
+
       // Cast `value` to a string to ensure the value is set correctly. While
       // browsers typically do this as necessary, jsdom doesn't.
-      DOMPropertyOperations.setValueForProperty(
-        ReactDOMComponentTree.getNodeFromInstance(inst),
-        'value',
-        '' + value
-      );
+      var newValue = '' + value;
+
+      // To avoid side effects (such as losing text selection), only set value if changed
+      if (newValue !== node.value) {
+        node.value = newValue;
+      }
+    } else {
+      if (props.value == null && props.defaultValue != null) {
+        invariant(inst._rootNodeID, 'Must be mounted to initialize initial input value');
+        node.defaultValue = '' + props.defaultValue;
+      }
+      if (props.checked == null && props.defaultChecked != null) {
+        invariant(inst._rootNodeID, 'Must be mounted to initialize initial input value');
+
+        // tmp is needed to work around a chrome bug where setting defaultChecked
+        // will sometimes influence the value of checked (even after detachment).
+        // Reference: https://bugs.chromium.org/p/chromium/issues/detail?id=608416
```

Comment:
Based on the patch that fixes it, it sounds like the issue is that setting .checked to its current value doesn't detach it. Maybe we can just do

```
node.checked = !node.checked;
node.checked = !node.checked;
```

in postMountWrapper with a link to the bug and/or commit?